beetsplug/discogs.py

```diff
@@ -205,11 +205,27 @@ def get_albums(self, query):
             self._log.debug(u"Communication error while searching for {0!r}",
                             query, exc_info=True)
             return []
-        return [self.get_album_info(release) for release in releases[:5]]
+        return filter(None, [self.get_album_info(release)
+                             for release in releases[:5]])
```

Comment:
Note that `filter` returns an iterable instead of a list on Python 3. That may be fine, but it's worth being aware. If a list is better for some reason, the canonical way is probably `[x for x in l if x]`.